app/flatpak-builtins-build.c

```diff
@@ -75,6 +75,82 @@ child_setup (gpointer user_data)
     fcntl (g_array_index (fd_array, int, i), F_SETFD, 0);
 }
 
+static gboolean
+find_matching_extension_group_in_metakey (GKeyFile    *metakey,
+                                          const char  *id,
+                                          const char  *specified_tag,
+                                          char       **out_extension_group,
+                                          GError     **error)
+{
+  g_auto(GStrv) groups = NULL;
+  g_autofree char *extension_prefix = NULL;
+  const char *last_seen_group = NULL;
+  guint n_extension_groups = 0;
+  GStrv iter = NULL;
+
+  g_return_val_if_fail (out_extension_group != NULL, FALSE);
+
+  groups =  g_key_file_get_groups (metakey, NULL);
+  extension_prefix = g_strconcat (FLATPAK_METADATA_GROUP_PREFIX_EXTENSION,
+                                  id,
+                                  NULL);
+
+  for (iter = groups; *iter != NULL; ++iter)
+    {
+      const char *group_name = *iter;
+      const char *extension_name = NULL;;
+      g_autofree char *extension_tag = NULL;
+
+      if (!g_str_has_prefix (group_name, extension_prefix))
+        continue;
+
+      ++n_extension_groups;
+      extension_name = group_name + strlen (FLATPAK_METADATA_GROUP_PREFIX_EXTENSION);
+
+      flatpak_parse_extension_with_tag (extension_name,
+                                        NULL,
+                                        &extension_tag);
+
+      /* Check 1: Does this extension have the same tag as the
+       * specified tag? If so, use it */
+      if (specified_tag != NULL &&
+          extension_tag != NULL &&
+          g_strcmp0 (extension_tag, specified_tag) == 0)
```

Comment:
This still seems wrong to me. It matches an exact tag, but if you have a non-tagged extension and one tagged there is no way to match the untagged one. I think this needs to be:
```
 if (g_strcmp0 (extension_tag, specified_tag) == 0)
```
I.e. if both specified_tag and extension_tag are NULL, we have an exact match.

Reply:
Makes sense. Updated.